openapi/src/parts/controlled_azure_database.yaml

```diff
@@ -0,0 +1,133 @@
+    get:
+      summary: Get a controlled Azure Database resource
+      operationId: getAzureDatabase
+      tags: [ ControlledAzureResource ]
+      responses:
+        '200':
+          description: Response to get database
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/AzureDatabaseResource'
+        '403':
+          $ref: '#/components/responses/PermissionDenied'
+        '404':
+          $ref: '#/components/responses/NotFound'
+        '500':
+          $ref: '#/components/responses/ServerError'
+    post:
+      summary: |
+        Delete a controlled Azure Database resource. This is async, because the 
+        operation can take a few minutes. OpenAPI does not support request body 
+        in DELETE, but async state requires it. Hence this is a POST.
+      operationId: deleteAzureDatabase
+      tags: [ ControlledAzureResource ]
+      requestBody:
+        required: true
+        content:
+          application/json:
+            schema:
+              $ref: '#/components/schemas/DeleteControlledAzureResourceRequest'
+      responses:
+        '200':
+          $ref: '#/components/responses/DeleteControlledAzureResourceResponse'
+        '202':
+          $ref: '#/components/responses/DeleteControlledAzureResourceResponse'
+        '403':
+          $ref: '#/components/responses/PermissionDenied'
+        '404':
+          $ref: '#components/responses/NotFound'
+        '500':
+          $ref: '#/components/responses/ServerError'
+
+
+  /api/workspaces/v1/{workspaceId}/resources/controlled/azure/databases/delete-result/{jobId}:
+    parameters:
+      - $ref: '#/components/parameters/WorkspaceId'
+      - $ref: '#/components/parameters/JobId'
+    get:
+      summary: Retrieve information about an azure database deletion job.
+      operationId: getDeleteAzureDatabaseResult
+      tags: [ ControlledAzureResource ]
+      responses:
+        '200':
+          $ref: '#/components/responses/DeleteControlledAzureResourceResponse'
+        '202':
+          $ref: '#/components/responses/DeleteControlledAzureResourceResponse'
+        '403':
+          $ref: '#/components/responses/PermissionDenied'
+        '404':
+          $ref: '#components/responses/NotFound'
+        '500':
+          $ref: '#/components/responses/ServerError'
+
+components:
+  schemas:
+    AzureDatabaseCreationParameters:
+      description: >-
+        Database-specific properties to be set on creation. These are a subset of the values
+        accepted by the azure resource API
+      type: object
+      required: [ name, owner ]
+      properties:
+        name:
+          description: A valid database name
+          type: string
+          pattern: ^[a-zA-Z][a-zA-Z0-9_]{0,31}$
+        owner:
+          description: Resource id of an azure managed identity in this workspace to be the owner of the database
+          type: string
+          format: uuid
+        k8sNamespace:
```

Comment:
Wondering if the k8s namespace _and_ the KSA should be input parameters here. Then Leo could reuse both for running apps.

Leo should probably also make sure the namespace/KSA "stick" to the database -- e.g. if a new app is connecting to an existing database, it should reuse the namespace/KSA associated with the database.

Reply:
Also: will WSM create the k8s namespace or does it expect it to be pre-created?

Reply:
right now WSM expects a pre-created namespace but in the future I would like WSM to own the namespace. I think WSM should own the KSA too.

Let's not worry about connecting to existing databases yet

Reply:
Should `k8sNamespace` be required?

Reply:
As implemented I think it's probably required -- it controls the namespace of the KSA linked to the ManagedIdentity who can access the database.